src/Autofac/RegistrationExtensions.Delegates.cs

```diff
@@ -0,0 +1,817 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Reflection;
+using Autofac.Builder;
+using Autofac.Core;
+using Autofac.Core.Activators.Delegate;
+using Autofac.Core.Activators.ProvidedInstance;
+using Autofac.Core.Activators.Reflection;
+using Autofac.Core.Lifetime;
+using Autofac.Core.Resolving.Pipeline;
+using Autofac.Features.Scanning;
+using Autofac.Util;
+
+namespace Autofac;
+
+/// <summary>
+/// Adds registration syntax to the <see cref="ContainerBuilder"/> type.
+/// </summary>
+[SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+public static partial class RegistrationExtensions
```

Comment:
I don't have a strong preference, but we had to split that one giant `RegistrationExtensions` class into files because we couldn't break binary compatibility and actually put them in different classes. Given these are net-new, is it worth actually having a different extensions class? I mean, _technically_ it doesn't matter, I guess, just thinking out loud.

Reply:
I think the reason they should be in the same class is that they have the same name (`Register`) as other extension methods, because of overload ambiguity detection; if the extension methods are in the same class, the compiler will add an error to the method definition that the overloads clash, but if they're in another class, the compiler will raise an "ambiguous" error on the caller of the method, which seems bad.

Reply:
Oh, that makes good sense. I didn't think of that.